### ISSUE TITLE
What is the difference between this version of FFMPEG and the official FFMPEG version?

### ISSUE THREAD

**Issue by curtwagner1984** [jellyfin/jellyfin-ffmpeg#84]
Hello, 

I'm trying to make a mini 'Just In Time' encoding application, meaning encode a video into `h264` as the user asks for it, and not before and stream it to a webpage. This is basically what JellyFin does on a larger scale.

I've been trying to do it with the regular `FFMPEG` but there's an issue when seeking.  The way I'm doing it now is starting to encode the video with FFMPEG and creating a DASH `m3u8` playlist. When I want to seek in the video, I turn FFMPEG off and restart it in the seek target timestamp. The problem with this is that the indexes of the parts FFMPEG creates after seeking are wrong. It basically starts from the beginning. For example, when FFMPEG starts it generates parts 0,1,2,3,4 but then I want to jump to part 20. I shut FFMPEG down, restart it with the timestamp for part 20, but when it actually generates the parts inside, the indexes are still 0,1,2 etc.

I've looked at this https://github.com/Dusk-Labs/nightfall repository by @vgarleanu that does JIT streaming with the regular FFMPEG. It involves (When seeking) manually opening the newly created `part` files and patching the correct index inside them.  I hope this can be avoided with this version of FFmpeg.

My hope is that this version of FFMPEG was specifically built for JIT streaming and it handles this issue internally. However, I was unable to find any documentation as to what is different in this version and how to use the new features. 

**TL;DR**
What are the new features in this FFMPEG compared to the original, and where can one find documentation on how to use them?

**Comment by cvium** [comment]
All of our non-build related changes are Debian-style patches https://github.com/jellyfin/jellyfin-ffmpeg/tree/jellyfin/debian/patches

We do not use DASH, so I'm afraid you won't see any benefits in this regard.

**Comment by curtwagner1984** [comment]
@cvium  How do you handle seeking if I may ask?

**Comment by cvium** [comment]
1. Check if part XYZ exists (if yes, go to Step 4)
2. Compute the start timestamp based on segment number
3. Restart ffmpeg with `-ss <start_time> -start_number XYZ`
4. Serve part XYZ

**Comment by curtwagner1984** [comment]
@cvium This works with the regular FFMPEG too? For some reason, I thought that FFMPEG puts an index number inside the actual XYZ part and not just in the `m3u8` playlist so when you restart it with a new start number, it only writes it in the playlist, while having the internal indexes in the parts themselves start at 0.

EDIT: And another question, how do you facilitate direct play? When you know that the file is already in h264. Do you just serve the file as is?

**Comment by cvium** [comment]
There's no index in the HLS segments themselves. There is just the DTS (decoding time stamp) and PTS (presentation time stamp). You can, if you want, set the DTS and PTS to 0 after seeking, but we obviously don't.

> EDIT: And another question, how do you facilitate direct play? When you know that the file is already in h264. Do you just serve the file as is?

The clients send a device profile and based on the profile and the video file's mediainfo we will serve it as-is.

**Comment by curtwagner1984** [comment]
Thank you. What's the difference between DASH and what you're using? Is the difference only in the part container? `mp4` vs `ts`?

**Comment by cvium** [comment]
I assume so but I have no experience with DASH.

**Comment by curtwagner1984** [comment]
> I assume so but I have no experience with DASH.

Thank you for your time answering questions. I'll try to do it as you suggested.
